apps/hash-graph/lib/graph/src/store/query/filter.rs

```diff
@@ -345,33 +345,31 @@ impl Parameter<'_> {
 
             // Number conversions
             (Parameter::Number(number), ParameterType::Any) => {
-                *self = Parameter::Any(Value::Number(Number::from_f64(*number).ok_or_else(
-                    || {
-                        Report::new(ParameterConversionError {
-                            actual: self.to_owned(),
-                            expected,
-                        })
-                    },
-                )?));
+                *self = Parameter::Any(Value::Number(Number::from(*number)));
             }
             (Parameter::Any(Value::Number(number)), ParameterType::Number) => {
-                *self = Parameter::Number(number.as_f64().ok_or_else(|| {
+                let number = number.as_i64().ok_or_else(|| {
                     Report::new(ParameterConversionError {
                         actual: self.to_owned(),
                         expected,
                     })
-                })?);
+                })?;
+                *self =
+                    Parameter::Number(i32::try_from(number).into_report().change_context_lazy(
```

Comment:
Not sure if I'm missing something, but are we not converting back and forth here?

Reply:
I.e. `i32 -> i64 -> i32`

Reply:
No, we have a `serde_json::Number`, which has to be parsed as either `i64`, `u64` or `f64`, so we're parsing it as `i64`.